app/src/display/widgets/output_status.c

```diff
@@ -36,62 +45,73 @@ void output_status_init() {
 }
 
-void set_status_symbol(lv_obj_t *label) {
-    enum zmk_endpoint selected_endpoint = zmk_endpoints_selected();
-    bool active_profile_connected = zmk_ble_active_profile_is_connected();
-    bool active_profie_bonded = !zmk_ble_active_profile_is_open();
-    uint8_t active_profile_index = zmk_ble_active_profile_index();
+static struct output_status_state get_state(const zmk_event_t *_eh) {
+    return (struct output_status_state){.selected_endpoint = zmk_endpoints_selected(),
+                                        .active_profile_connected =
+                                            zmk_ble_active_profile_is_connected(),
+                                        .active_profile_bonded = !zmk_ble_active_profile_is_open(),
+                                        .active_profile_index = zmk_ble_active_profile_index()};
+    ;
+}
+
+static void set_status_symbol(lv_obj_t *label, struct output_status_state state) {
     char text[6] = {};
 
-    switch (selected_endpoint) {
+    switch (state.selected_endpoint) {
     case ZMK_ENDPOINT_USB:
         strcat(text, LV_SYMBOL_USB "   ");
         break;
     case ZMK_ENDPOINT_BLE:
-        if (active_profie_bonded) {
-            if (active_profile_connected) {
-                sprintf(text, LV_SYMBOL_WIFI "%i " LV_SYMBOL_OK, active_profile_index);
+        if (state.active_profile_bonded) {
+            if (state.active_profile_connected) {
+                snprintf(text, sizeof(text), LV_SYMBOL_WIFI "%i " LV_SYMBOL_OK,
+                         state.active_profile_index);
             } else {
-                sprintf(text, LV_SYMBOL_WIFI "%i " LV_SYMBOL_CLOSE, active_profile_index);
+                snprintf(text, sizeof(text), LV_SYMBOL_WIFI "%i " LV_SYMBOL_CLOSE,
+                         state.active_profile_index);
             }
         } else {
-            sprintf(text, LV_SYMBOL_WIFI "%i " LV_SYMBOL_SETTINGS, active_profile_index);
+            snprintf(text, sizeof(text), LV_SYMBOL_WIFI "%i " LV_SYMBOL_SETTINGS,
+                     state.active_profile_index);
         }
         break;
     }
 
     lv_label_set_text(label, text);
 }
 
+static void output_status_update_cb(struct output_status_state state) {
+    struct zmk_widget_output_status *widget;
+    SYS_SLIST_FOR_EACH_CONTAINER(&widgets, widget, node) { set_status_symbol(widget->obj, state); }
+}
+
+ZMK_DISPLAY_WIDGET_LISTENER(widget_output_status, struct output_status_state,
+                            output_status_update_cb, get_state)
+ZMK_SUBSCRIPTION(widget_output_status, zmk_endpoint_selection_changed);
+
+#if defined(CONFIG_USB)
+ZMK_SUBSCRIPTION(widget_output_status, zmk_usb_conn_state_changed);
+#endif
+#if defined(CONFIG_ZMK_BLE)
+ZMK_SUBSCRIPTION(widget_output_status, zmk_ble_active_profile_changed);
+#endif
+
 int zmk_widget_output_status_init(struct zmk_widget_output_status *widget, lv_obj_t *parent) {
     output_status_init();
+    // widget_output_status_refresh_state(NULL);
```

Comment:
Are these intended to be left commented out?

Reply:
Nope, removed.